### DISCUSSION TITLE
[general] Some lints to fix in the package

### DISCUSSION THREAD

**Discussion by etiennebacher** [ropensci/targets#1321] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hello, I'm building a new package to find and automatically fix lints in R code: [`flint`](https://flint.etiennebacher.com/).

I'm using real-life, large packages to check its performance and `targets` is one of them. Since I already test on this, proposing those changes doesn't add much cost. I ran `flint::fix_package(exclude_linters = "outer_negation")` and fixed a wrong replacement by hand. The changes can be seen here: https://github.com/ropensci/targets/compare/main...etiennebacher:targets:main

There are many changes but most of them are trivial:
- use `inherits()` instead of  `"foo" %in% class(x)`
- use `expect_named()`, `expect_null()`, and other variants in tests
etc.

`flint` is quite new and linter rules are necessarily a bit opinionated, so some of those changes might not be to your taste, but I'd be happy to have some feedback on this. If you are ok with this idea, then I can open a PR.

**Comment by wlandau** [comment]
Sure, happy to review a PR.
